streaming_providers/debrid_client.py

```
@@ -1,121 +1,127 @@
-    def __del__(self):
+    async def __aenter__(self):
+        await self.initialize_headers()
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
         if self.token:
             try:
-                self.disable_access_token()
+                await self.disable_access_token()
             except ProviderException:
                 pass
```

Ruff

Use `contextlib.suppress(ProviderException)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(ProviderException)`

(SIM105)